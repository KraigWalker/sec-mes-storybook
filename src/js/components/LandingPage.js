import React from 'react';
import { Link } from 'react-router-dom';
import { getActiveTab } from '../actions/AppActions';
import { connect } from 'react-redux';
import { utils } from "document-management-web-ui";

import SecureMessageTabs from './SecureMessageTabs';
import StepHeader from './common/StepHeader';
import { SecureMessageBL } from '../bl/SecureMessageBL'
import GetIcon from './common/GetIcon';
import ErrorPage from './common/ErrorPage';
import SvgIcon from './common/GetIcon.js';
const Fingerprint = require('fingerprintjs2');

/**
 * @class Landing Page
 * Landing Page of the application
*/

export class LandingPage extends React.PureComponent {
    componentDidMount() {
        window.top.postMessage('clearNewMessagePage', '*');
        window.scrollTo(0, 0);
    }

    linkClick = activeTab => {
        this.props.dispatch(getActiveTab(activeTab));
    }

    handleBackClick = () => {
        window.top.postMessage('goBackToAccount', '*');
    }
    mapMessages(messages) {
        return SecureMessageBL(messages);
    }
    checkError() {
        const { isWebView } = this.props;
        if (this.props.messages.error && this.props.messages.fetched) {
            this.props.history.push('/errormessage');
        } else {
            return (
                <div className="row centralised-container">
                    <div className="col-md1-24 col-sm1-24 col-lg1-24">
                    { !isWebView &&
                        <p className="c-step-header__crumbs">
                            <a onClick={this.handleBackClick} className="c-step-header__link u-cursor-pointer">
                                <span className="c-step-header__linkicon"><SvgIcon id="icon-left" width="16px" height="16px" /></span>
                                <span className="c-step-header__linktext">{this.props.content.backToAccounts}</span>
                            </a>
                        </p>
                        }
                        <h1 className="c-step-header__title" id="headingTag" tabIndex="-1">{this.props.content.messages}</h1>
                        <p className="c-step-header__subtext">{this.props.content.landingPageMessage}</p>
                        <p className="c-step-header__subtext">{this.props.content.faqLink}</p>
                        <Link className="c-btn c-btn--default u-margin-bottom-c new-message-btn" to={{ pathname: `${window.baseURl}/securemessages/new` }}>
                            <GetIcon id="icon-pencil" width="16px" height="16px" />{this.props.content.newSecureMessage}
                        </Link>
                        <SecureMessageTabs location={this.props.location} onClick={this.linkClick} messages={this.mapMessages(this.props.messages)} activeTab={this.props.activeTab} content={this.props.content} />
                    </div>
                </div>
            );
        }
    }
    render() {
        return this.checkError();
    }
}
/**
 * Maps the state of the component to the state of the redux store
 * @param {object} state. State of the application
 */

const mapState = state => ({
    messages: state.messages,
    accounts: state.accounts,
    activeTab: state.messages.activeTab,
});

console.log(utils, utils.withNativeBridge(window.navigator.userAgent)(LandingPage))

export default connect(mapState)(utils.withNativeBridge(window.navigator.userAgent)(LandingPage));
