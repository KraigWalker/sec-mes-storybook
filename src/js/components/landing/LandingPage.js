import React from 'react';
import { Link } from 'react-router-dom';
import { getSecureMessages, getActiveTab } from '../../actions/AppActions';
import { connect } from 'react-redux';
import SecureMessageTabs from './SecureMessageTabs';
import StepHeader from '../common/StepHeader';
/**
 * @class Landing Page 
 * Landing Page of the application
*/

class LandingPage extends React.PureComponent {
    constructor(props) {
        super(props);
    }
    componentWillMount() {
        !this.props.messages.fetched && this.props.dispatch(getSecureMessages());
    }
    linkClick = (activeTab) => {
        this.props.dispatch(getActiveTab(activeTab));
    }
    render() {
        return(
            <div>
                <StepHeader showheaderCrumbs={false} onClick={() => {}} headerCrumbsMessage="" headerTitle={this.props.headerDetails.title} headerSubtext={this.props.headerDetails.subtext}/>
                <Link to = {{ pathname : '/newsecuremessage' }}><input type = 'button' name = 'New secure message' value = 'New secure message'/></Link>
                <SecureMessageTabs location={this.props.location} onClick={this.linkClick}  messages={this.props.messages} activeTab={this.props.activeTab}/>
            </div>
        );
    }
}
/**
 * Maps the state of the component to the state of the redux store
 * @param {object} state. State of the application
 */

const mapState = (state) => { 
    return {
        messages: state.messages,
    }
};


export default connect(mapState)(LandingPage);