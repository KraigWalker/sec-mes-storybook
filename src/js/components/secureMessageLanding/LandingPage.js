import React from 'react';
import { Link } from 'react-router-dom';
import { fetchSecureMessages, getActiveTab } from '../../actions/AppActions';
import { connect } from 'react-redux';
import SecureMessageTabs from './SecureMessageTabs';
import StepHeader from '../common/StepHeader';
import {SecureMessageBL} from '../../bl/SecureMessageBL'
/**
 * @class Landing Page 
 * Landing Page of the application
*/

class LandingPage extends React.PureComponent {
    constructor(props) {
        super(props);
    }
    componentWillMount() {
        !this.props.messages.fetched && this.props.dispatch(fetchSecureMessages());
    }
    linkClick = (activeTab) => {
        this.props.dispatch(getActiveTab(activeTab));
    }
    mapMessages(messages){
        return SecureMessageBL(messages);
    }
    render() {
        return(
            <div className="container">
                <div className="row">
                    <div className="col-md1-18">
                        <StepHeader showheaderCrumbs={false} headerTitle={this.props.content.landingPageTitle} headerSubtext={this.props.content.landingPageMessage}/>
                    </div>
                </div>
                <Link className="c-btn c-btn--default" to = {{ pathname : '/newsecuremessage' }}>New secure message</Link>
                <SecureMessageTabs location={this.props.location} onClick={this.linkClick}  messages={this.mapMessages(this.props.messages)} activeTab={this.props.activeTab}/>
            </div>
        );
    }
}
/**
 * Maps the state of the component to the state of the redux store
 * @param {object} state. State of the application
 */

const mapState = (state) => { 
    return {
        messages: state.messages,
        activeTab: state.messages.activeTab,
    }
};


export default connect(mapState)(LandingPage);