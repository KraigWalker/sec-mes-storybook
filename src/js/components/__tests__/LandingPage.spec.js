import React from 'react';
import { shallow, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import TestUtils from 'react-addons-test-utils';
import { createMockStore } from 'redux-test-utils';
import { LandingPage } from '../LandingPage';
import { Provider } from "react-redux";

let props;
let state;
describe("Landing Page else snapshot", () => {
    const dispatch = jest.fn();
    const linkClick = jest.fn();
    const postMessage = jest.fn();
    let props = {
        content: {
            back: 'Back',
            retry: 'Retry',
            backToAccounts: 'Back to accounts'
        },
        messages: {
            error: false,
            fetched: false
        },
        dispatch: dispatch
    };

    let component = shallow(<LandingPage {...props} />);
    it('should match to snapshot', () => {
        expect(component).toMatchSnapshot();
    });
});
describe("Landing Page if snapshot", () => {
    const dispatch = jest.fn();
    const push = jest.fn();
    let props = {
        content: {
            back: 'Back',
            retry: 'Retry',
            backToAccounts: 'Back to accounts'
        },
        messages: {
            error: true,
            fetched: true
        },
        history: {
            push: push
        },
        dispatch: dispatch
    };
    let component = shallow(<LandingPage {...props} />);
    it('should match to snapshot', () => {
        expect(push).toBeCalled();
    });
});
describe("Landing Page if snapshot", () => {
    it('should call link clicked method', () => {
        const dispatch = jest.fn();
        let props = {
            content: {

            },
            messages: {
                error: false,
                fetched: false
            },
            dispatch: dispatch
        };
        const wrapper = shallow(
            <LandingPage {...props} />
        );
        wrapper.find('SecureMessageTabs').simulate('click');
    });
});
describe("Landing Page if snapshot", () => {
    it('should call handle back clicked method', () => {
        const dispatch = jest.fn();
        let props = {
            content: {

            },
            messages: {
                error: false,
                fetched: false
            },
            dispatch: dispatch
        };
        const wrapper = shallow(
            <LandingPage {...props} />
        );
        wrapper.find('a').simulate('click');
       // expect(getActiveTab).toBeCalled();
    });
});