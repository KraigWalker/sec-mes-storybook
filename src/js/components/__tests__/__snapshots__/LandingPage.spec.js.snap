// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing Page else snapshot should match to snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LandingPage
    content={
      Object {
        "back": "Back",
        "backToAccounts": "Back to accounts",
        "retry": "Retry",
      }
    }
    dispatch={[MockFunction]}
    fingerPrintID="2344323434523"
    getActiveTab={[MockFunction]}
    getMessageSubjects={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
      }
    }
    messages={
      Object {
        "error": false,
        "fetched": false,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <Row>
        <Card
          bordered={false}
        >
          <TextBody
            className="c-step-header__crumbs"
          >
            <BackButton
              disabled={false}
              label="Back to accounts"
              onClick={[Function]}
            />
          </TextBody>
          <Title
            size="h1"
          />
          <TextBody />
          <TextBody />
          <Button
            disabled={false}
            display="primary"
            isLoading={false}
            onClick={[Function]}
            size="md"
          />
          <SecureMessageTabs
            activeTab={undefined}
            content={
              Object {
                "back": "Back",
                "backToAccounts": "Back to accounts",
                "retry": "Retry",
              }
            }
            location={undefined}
            messages={
              Object {
                "archivedMessages": Array [],
                "draftMessages": Array [],
                "inboxMessages": Array [],
                "sentMessages": Array [],
              }
            }
            onClick={[Function]}
          />
        </Card>
      </Row>,
      "className": "u-margin-top-6",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Card
          bordered={false}
        >
          <TextBody
            className="c-step-header__crumbs"
          >
            <BackButton
              disabled={false}
              label="Back to accounts"
              onClick={[Function]}
            />
          </TextBody>
          <Title
            size="h1"
          />
          <TextBody />
          <TextBody />
          <Button
            disabled={false}
            display="primary"
            isLoading={false}
            onClick={[Function]}
            size="md"
          />
          <SecureMessageTabs
            activeTab={undefined}
            content={
              Object {
                "back": "Back",
                "backToAccounts": "Back to accounts",
                "retry": "Retry",
              }
            }
            location={undefined}
            messages={
              Object {
                "archivedMessages": Array [],
                "draftMessages": Array [],
                "inboxMessages": Array [],
                "sentMessages": Array [],
              }
            }
            onClick={[Function]}
          />
        </Card>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "bordered": false,
          "children": Array [
            <TextBody
              className="c-step-header__crumbs"
            >
              <BackButton
                disabled={false}
                label="Back to accounts"
                onClick={[Function]}
              />
            </TextBody>,
            <Title
              size="h1"
            />,
            <TextBody />,
            <TextBody />,
            <Button
              disabled={false}
              display="primary"
              isLoading={false}
              onClick={[Function]}
              size="md"
            />,
            <SecureMessageTabs
              activeTab={undefined}
              content={
                Object {
                  "back": "Back",
                  "backToAccounts": "Back to accounts",
                  "retry": "Retry",
                }
              }
              location={undefined}
              messages={
                Object {
                  "archivedMessages": Array [],
                  "draftMessages": Array [],
                  "inboxMessages": Array [],
                  "sentMessages": Array [],
                }
              }
              onClick={[Function]}
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <BackButton
                disabled={false}
                label="Back to accounts"
                onClick={[Function]}
              />,
              "className": "c-step-header__crumbs",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "disabled": false,
                "label": "Back to accounts",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": undefined,
              "size": "h1",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": undefined,
              "disabled": false,
              "display": "primary",
              "isLoading": false,
              "onClick": [Function],
              "size": "md",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "activeTab": undefined,
              "content": Object {
                "back": "Back",
                "backToAccounts": "Back to accounts",
                "retry": "Retry",
              },
              "location": undefined,
              "messages": Object {
                "archivedMessages": Array [],
                "draftMessages": Array [],
                "inboxMessages": Array [],
                "sentMessages": Array [],
              },
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Row>
          <Card
            bordered={false}
          >
            <TextBody
              className="c-step-header__crumbs"
            >
              <BackButton
                disabled={false}
                label="Back to accounts"
                onClick={[Function]}
              />
            </TextBody>
            <Title
              size="h1"
            />
            <TextBody />
            <TextBody />
            <Button
              disabled={false}
              display="primary"
              isLoading={false}
              onClick={[Function]}
              size="md"
            />
            <SecureMessageTabs
              activeTab={undefined}
              content={
                Object {
                  "back": "Back",
                  "backToAccounts": "Back to accounts",
                  "retry": "Retry",
                }
              }
              location={undefined}
              messages={
                Object {
                  "archivedMessages": Array [],
                  "draftMessages": Array [],
                  "inboxMessages": Array [],
                  "sentMessages": Array [],
                }
              }
              onClick={[Function]}
            />
          </Card>
        </Row>,
        "className": "u-margin-top-6",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Card
            bordered={false}
          >
            <TextBody
              className="c-step-header__crumbs"
            >
              <BackButton
                disabled={false}
                label="Back to accounts"
                onClick={[Function]}
              />
            </TextBody>
            <Title
              size="h1"
            />
            <TextBody />
            <TextBody />
            <Button
              disabled={false}
              display="primary"
              isLoading={false}
              onClick={[Function]}
              size="md"
            />
            <SecureMessageTabs
              activeTab={undefined}
              content={
                Object {
                  "back": "Back",
                  "backToAccounts": "Back to accounts",
                  "retry": "Retry",
                }
              }
              location={undefined}
              messages={
                Object {
                  "archivedMessages": Array [],
                  "draftMessages": Array [],
                  "inboxMessages": Array [],
                  "sentMessages": Array [],
                }
              }
              onClick={[Function]}
            />
          </Card>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "bordered": false,
            "children": Array [
              <TextBody
                className="c-step-header__crumbs"
              >
                <BackButton
                  disabled={false}
                  label="Back to accounts"
                  onClick={[Function]}
                />
              </TextBody>,
              <Title
                size="h1"
              />,
              <TextBody />,
              <TextBody />,
              <Button
                disabled={false}
                display="primary"
                isLoading={false}
                onClick={[Function]}
                size="md"
              />,
              <SecureMessageTabs
                activeTab={undefined}
                content={
                  Object {
                    "back": "Back",
                    "backToAccounts": "Back to accounts",
                    "retry": "Retry",
                  }
                }
                location={undefined}
                messages={
                  Object {
                    "archivedMessages": Array [],
                    "draftMessages": Array [],
                    "inboxMessages": Array [],
                    "sentMessages": Array [],
                  }
                }
                onClick={[Function]}
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <BackButton
                  disabled={false}
                  label="Back to accounts"
                  onClick={[Function]}
                />,
                "className": "c-step-header__crumbs",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "disabled": false,
                  "label": "Back to accounts",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": undefined,
                "size": "h1",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": undefined,
                "disabled": false,
                "display": "primary",
                "isLoading": false,
                "onClick": [Function],
                "size": "md",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "activeTab": undefined,
                "content": Object {
                  "back": "Back",
                  "backToAccounts": "Back to accounts",
                  "retry": "Retry",
                },
                "location": undefined,
                "messages": Object {
                  "archivedMessages": Array [],
                  "draftMessages": Array [],
                  "inboxMessages": Array [],
                  "sentMessages": Array [],
                },
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenFourAdapter {
      "options": Object {
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "prevContext": true,
          },
          "getChildContext": Object {
            "calledByRenderer": true,
          },
        },
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
